File: apple/Sources/CCDAEngine/Models/CCDAValue.swift
import Foundation

public struct CCDAValue: Hashable {
    public let type: String?
    public let code: String?
    public let displayName: String?
    public let value: String?
    public let unit: String?

    public var display: String {
        if let displayName {
            return displayName
        }

        return [value, unit].compactMap { $0 }.joined(separator: " ")
    }
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Common/CCDAValue.kt
package com.ccdakit.engine

/**
 * Value parsed from an entry `value` element.
 *
 * @property type XML or `xsi:type` value such as `CD` or `PQ`.
 * @property code Coded value when the value is coded.
 * @property displayName Human-readable coded display.
 * @property value Scalar value when the value is numeric or textual.
 * @property unit Unit for scalar values.
 */
data class CCDAValue(
    val type: String? = null,
    val code: String? = null,
    val displayName: String? = null,
    val value: String? = null,
    val unit: String? = null,
)
